From 4eb7c63aeec95da5854a095983e055e1700f4f77 Mon Sep 17 00:00:00 2001
From: Amir Vadai <avadai@redhat.com>
Date: Thu, 11 Sep 2014 11:59:00 -0400
Subject: [lib] cpumask: cpumask_set_cpu_local_first to use all cores when numa node is not defined

Message-id: <00e74dfac296e2ab1f9c5a22491ba3e148da4b77.1410425016.git.avadai@redhat.com>
Patchwork-id: 93582
O-Subject: [Patch RHEL7.1 V1 004/197] lib/cpumask: cpumask_set_cpu_local_first to use all cores when numa node is not defined
Bugzilla: 1107617 1107618 1107619 1107620
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

From: Amir Vadai <amirv@mellanox.com>

When device is non numa aware (numa_node == -1), use all online cpu's.

Signed-off-by: Amir Vadai <amirv@mellanox.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit 143b5ba21b2bd5091cd8dcd92de7ba1ed1d1c83c)
Signed-off-by: Amir Vadai <amirv@mellanox.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/lib/cpumask.c b/lib/cpumask.c
index be5f391..c8d357a 100644
--- a/lib/cpumask.c
+++ b/lib/cpumask.c
@@ -191,7 +191,7 @@ int cpumask_set_cpu_local_first(int i, int numa_node, cpumask_t *dstp)
 
  i %= num_online_cpus();
 
- if (!cpumask_of_node(numa_node)) {
+ if (numa_node == -1 || !cpumask_of_node(numa_node)) {
   /* Use all online cpu's for non numa aware system */
   cpumask_copy(mask, cpu_online_mask);
  } else {
-- 
1.7.1